eva/catalog/catalog_utils.py

```diff
@@ -83,3 +101,49 @@ def xform_column_definitions_to_catalog_entries(
         result_list.append(column_entry)
 
     return result_list
+
+
+def construct_udf_cache_catalog_entry(
+    func_expr: FunctionExpression,
+) -> UdfCacheCatalogEntry:
+    """Constructs a udf cache catalog entry from a given function expression.
+    It is assumed that the function expression has already been bound using the binder.
+    The catalog entry is populated with dependent udfs and columns by traversing the
+    expression tree. The cache name is represented by the signature of the function
+    expression.
+    Args:
+        func_expr (FunctionExpression): the function expression with which the cache is
+        assoicated
+    Returns:
+        UdfCacheCatalogEntry: the udf cache catalog entry
+    """
+    udf_depends = []
+    col_depends = []
+    for expr in func_expr.find_all(FunctionExpression):
+        udf_depends.append(expr.udf_obj.row_id)
+    for expr in func_expr.find_all(TupleValueExpression):
+        col_depends.append(expr.col_object.row_id)
+    cache_name = func_expr.signature()
+
+    # add salt to the cache_name so that we generate unique name
+    path = str(get_str_hash(cache_name + uuid.uuid4().hex))
+    cache_dir = ConfigurationManager().get_value("storage", "cache_dir")
+    cache_path = str(Path(cache_dir) / Path(f"{path}_{func_expr.name}"))
+    args = tuple([arg.signature() for arg in func_expr.children])
+    entry = UdfCacheCatalogEntry(
+        name=func_expr.signature(),
+        udf_id=func_expr.udf_obj.row_id,
+        cache_path=cache_path,
+        args=args,
+        udf_depends=udf_depends,
+        col_depends=col_depends,
```

Comment:
What is the motivation to store all this information instead of just func_expr's signature?

Reply:
`udf_id`: to know which udf we are building cache on. Can be derived from the signature but it will add unnecessary bugs. 
`cache_path`: to know the path
`args`: this can be removed 
`udf_depends` and `col_depends`: required to know when to dorp cache. If any of the dependant col or udf is dropped, we removed the cache.

Reply:
Got.